### PR TITLE
Fix issue with windows's sys.executable contains spaces

### PR DESCRIPTION
**Pull request by ewongbb**
This 'fixes' Issue #3733.  

Though I'm not sure I'm going about in the right manner.

### CODE REVIEW

**Comment by ewongbb** [comment]
What I'm trying to 'parse' for the first command and then to check whether or not it has spaces
in it and if it doesn't have quotes already, we add the quotes for the command.  Since this is my first foray 
into this part of the code, I'm kinda confused as to what the real command is and what the fake command is.  And whether or not this approach is even workable.

**Comment by tardyp** [comment]
please don't refer to trac issues, and avoid requiring to look at the issue to know what is being fixed

**Comment by tardyp** [comment]
I think the problem is around that line:

https://github.com/buildbot/buildbot/blob/master/worker/buildbot_worker/runprocess.py#L499

                display = shell_quote(self.fake_command, self.builder.unicode_encoding)

Looks like at some point, sombody wanted to shell quote the command for windows, but at some other point, somebody only kept the quoting for fake_display (which makes few sense)

anyway, we should probably keep the solution to this problem to the "if windows" block I just pointed

**Comment by ewongbb** [comment]
fwiw.. it wasn't a trac issue that I mentioned. It was a github one.

**Comment by tardyp** [comment]
> fwiw.. it wasn't a trac issue that I mentioned. It was a github one.

Sorry for that grumpiness of mine :) You can always link a github issue by using a '#' before the number.

Anyway, you did not address the other more important remark.
https://github.com/buildbot/buildbot/pull/3734#issuecomment-341705554

Your implementation should be windows specific I think. It will probably fail on linux if somebody is using space in its virtualenv (which we do not test), because linux does not expect the quote its paths.

real_command is the command that is actually executed, and fake_command is the one that is used for any logging. That one is supposed to be cleared from any passwords or secrets.

Please have a look at this shell_quote function I was pointing to you. I think this is the solution, rather than implementing your own.

File: worker/buildbot_worker/util/__init__.py [hunk 1]
```diff
@@ -74,6 +74,30 @@ def to_text(s):
             return s
         return str(s)
 
+    @staticmethod
```
Comment (tardyp) [inline worker/buildbot_worker/util/__init__.py]: I think this is not needed anymore

**Pull request closed without merging**